<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <meta name="generator" content="pandoc">
  <meta name="author" content="M Hallett" />
  <title>Transcription factor binding sites in Baker’s yeast</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="13-slides_files/reveal.js-3.3.0.1/css/reveal.css"/>



<link rel="stylesheet" href="13-slides_files/reveal.js-3.3.0.1/css/theme/black.css" id="theme">

<style type="text/css">
.reveal section img {
  background: rgba(255, 255, 255, 0.85);
}
</style>

  <!-- some tweaks to reveal css -->
  <style type="text/css">
    .reveal h1 { font-size: 2.0em; }
    .reveal h2 { font-size: 1.5em;  }
    .reveal h3 { font-size: 1.25em;	}
    .reveal h4 { font-size: 1em;	}

    .reveal .slides>section,
    .reveal .slides>section>section {
      padding: 0px 0px;
    }



    .reveal table {
      border-width: 1px;
      border-spacing: 2px;
      border-style: dotted;
      border-color: gray;
      border-collapse: collapse;
      font-size: 0.7em;
    }

    .reveal table th {
      border-width: 1px;
      padding-left: 10px;
      padding-right: 25px;
      font-weight: bold;
      border-style: dotted;
      border-color: gray;
    }

    .reveal table td {
      border-width: 1px;
      padding-left: 10px;
      padding-right: 25px;
      border-style: dotted;
      border-color: gray;
    }

    .reveal .slide-menu-button {
      left: 105px !important;
    }

  </style>

    <style type="text/css">code{white-space: pre;}</style>

    <link rel="stylesheet" href="solar.css"/>

<!-- Printing and PDF exports -->
<script id="paper-css" type="application/dynamic-css">

/* Default Print Stylesheet Template
   by Rob Glazebrook of CSSnewbie.com
   Last Updated: June 4, 2008

   Feel free (nay, compelled) to edit, append, and
   manipulate this file as you see fit. */


@media print {

	/* SECTION 1: Set default width, margin, float, and
	   background. This prevents elements from extending
	   beyond the edge of the printed page, and prevents
	   unnecessary background images from printing */
	html {
		background: #fff;
		width: auto;
		height: auto;
		overflow: visible;
	}
	body {
		background: #fff;
		font-size: 20pt;
		width: auto;
		height: auto;
		border: 0;
		margin: 0 5%;
		padding: 0;
		overflow: visible;
		float: none !important;
	}

	/* SECTION 2: Remove any elements not needed in print.
	   This would include navigation, ads, sidebars, etc. */
	.nestedarrow,
	.controls,
	.fork-reveal,
	.share-reveal,
	.state-background,
	.reveal .progress,
	.reveal .backgrounds {
		display: none !important;
	}

	/* SECTION 3: Set body font face, size, and color.
	   Consider using a serif font for readability. */
	body, p, td, li, div {
		font-size: 20pt!important;
		font-family: Georgia, "Times New Roman", Times, serif !important;
		color: #000;
	}

	/* SECTION 4: Set heading font face, sizes, and color.
	   Differentiate your headings from your body text.
	   Perhaps use a large sans-serif for distinction. */
	h1,h2,h3,h4,h5,h6 {
		color: #000!important;
		height: auto;
		line-height: normal;
		font-family: Georgia, "Times New Roman", Times, serif !important;
		text-shadow: 0 0 0 #000 !important;
		text-align: left;
		letter-spacing: normal;
	}
	/* Need to reduce the size of the fonts for printing */
	h1 { font-size: 28pt !important;  }
	h2 { font-size: 24pt !important; }
	h3 { font-size: 22pt !important; }
	h4 { font-size: 22pt !important; font-variant: small-caps; }
	h5 { font-size: 21pt !important; }
	h6 { font-size: 20pt !important; font-style: italic; }

	/* SECTION 5: Make hyperlinks more usable.
	   Ensure links are underlined, and consider appending
	   the URL to the end of the link for usability. */
	a:link,
	a:visited {
		color: #000 !important;
		font-weight: bold;
		text-decoration: underline;
	}
	/*
	.reveal a:link:after,
	.reveal a:visited:after {
		content: " (" attr(href) ") ";
		color: #222 !important;
		font-size: 90%;
	}
	*/


	/* SECTION 6: more reveal.js specific additions by @skypanther */
	ul, ol, div, p {
		visibility: visible;
		position: static;
		width: auto;
		height: auto;
		display: block;
		overflow: visible;
		margin: 0;
		text-align: left !important;
	}
	.reveal pre,
	.reveal table {
		margin-left: 0;
		margin-right: 0;
	}
	.reveal pre code {
		padding: 20px;
		border: 1px solid #ddd;
	}
	.reveal blockquote {
		margin: 20px 0;
	}
	.reveal .slides {
		position: static !important;
		width: auto !important;
		height: auto !important;

		left: 0 !important;
		top: 0 !important;
		margin-left: 0 !important;
		margin-top: 0 !important;
		padding: 0 !important;
		zoom: 1 !important;

		overflow: visible !important;
		display: block !important;

		text-align: left !important;
		-webkit-perspective: none;
		   -moz-perspective: none;
		    -ms-perspective: none;
		        perspective: none;

		-webkit-perspective-origin: 50% 50%;
		   -moz-perspective-origin: 50% 50%;
		    -ms-perspective-origin: 50% 50%;
		        perspective-origin: 50% 50%;
	}
	.reveal .slides section {
		visibility: visible !important;
		position: static !important;
		width: auto !important;
		height: auto !important;
		display: block !important;
		overflow: visible !important;

		left: 0 !important;
		top: 0 !important;
		margin-left: 0 !important;
		margin-top: 0 !important;
		padding: 60px 20px !important;
		z-index: auto !important;

		opacity: 1 !important;

		page-break-after: always !important;

		-webkit-transform-style: flat !important;
		   -moz-transform-style: flat !important;
		    -ms-transform-style: flat !important;
		        transform-style: flat !important;

		-webkit-transform: none !important;
		   -moz-transform: none !important;
		    -ms-transform: none !important;
		        transform: none !important;

		-webkit-transition: none !important;
		   -moz-transition: none !important;
		    -ms-transition: none !important;
		        transition: none !important;
	}
	.reveal .slides section.stack {
		padding: 0 !important;
	}
	.reveal section:last-of-type {
		page-break-after: avoid !important;
	}
	.reveal section .fragment {
		opacity: 1 !important;
		visibility: visible !important;

		-webkit-transform: none !important;
		   -moz-transform: none !important;
		    -ms-transform: none !important;
		        transform: none !important;
	}
	.reveal section img {
		display: block;
		margin: 15px 0px;
		background: rgba(255,255,255,1);
		border: 1px solid #666;
		box-shadow: none;
	}

	.reveal section small {
		font-size: 0.8em;
	}

}  
</script>


<script id="pdf-css" type="application/dynamic-css">
    
/**
 * This stylesheet is used to print reveal.js
 * presentations to PDF.
 *
 * https://github.com/hakimel/reveal.js#pdf-export
 */

* {
	-webkit-print-color-adjust: exact;
}

body {
	margin: 0 auto !important;
	border: 0;
	padding: 0;
	float: none !important;
	overflow: visible;
}

html {
	width: 100%;
	height: 100%;
	overflow: visible;
}

/* Remove any elements not needed in print. */
.nestedarrow,
.reveal .controls,
.reveal .progress,
.reveal .playback,
.reveal.overview,
.fork-reveal,
.share-reveal,
.state-background {
	display: none !important;
}

h1, h2, h3, h4, h5, h6 {
	text-shadow: 0 0 0 #000 !important;
}

.reveal pre code {
	overflow: hidden !important;
	font-family: Courier, 'Courier New', monospace !important;
}

ul, ol, div, p {
	visibility: visible;
	position: static;
	width: auto;
	height: auto;
	display: block;
	overflow: visible;
	margin: auto;
}
.reveal {
	width: auto !important;
	height: auto !important;
	overflow: hidden !important;
}
.reveal .slides {
	position: static;
	width: 100%;
	height: auto;

	left: auto;
	top: auto;
	margin: 0 !important;
	padding: 0 !important;

	overflow: visible;
	display: block;

	-webkit-perspective: none;
	   -moz-perspective: none;
	    -ms-perspective: none;
	        perspective: none;

	-webkit-perspective-origin: 50% 50%; /* there isn't a none/auto value but 50-50 is the default */
	   -moz-perspective-origin: 50% 50%;
	    -ms-perspective-origin: 50% 50%;
	        perspective-origin: 50% 50%;
}

.reveal .slides section {
	page-break-after: always !important;

	visibility: visible !important;
	position: relative !important;
	display: block !important;
	position: relative !important;

	margin: 0 !important;
	padding: 0 !important;
	box-sizing: border-box !important;
	min-height: 1px;

	opacity: 1 !important;

	-webkit-transform-style: flat !important;
	   -moz-transform-style: flat !important;
	    -ms-transform-style: flat !important;
	        transform-style: flat !important;

	-webkit-transform: none !important;
	   -moz-transform: none !important;
	    -ms-transform: none !important;
	        transform: none !important;
}

.reveal section.stack {
	margin: 0 !important;
	padding: 0 !important;
	page-break-after: avoid !important;
	height: auto !important;
	min-height: auto !important;
}

.reveal img {
	box-shadow: none;
}

.reveal .roll {
	overflow: visible;
	line-height: 1em;
}

/* Slide backgrounds are placed inside of their slide when exporting to PDF */
.reveal section .slide-background {
	display: block !important;
	position: absolute;
	top: 0;
	left: 0;
	width: 100%;
	z-index: -1;
}

/* All elements should be above the slide-background */
.reveal section>* {
	position: relative;
	z-index: 1;
}

/* Display slide speaker notes when 'showNotes' is enabled */
.reveal .speaker-notes-pdf {
	display: block;
	width: 100%;
	max-height: none;
	left: auto;
	top: auto;
	z-index: 100;
}

/* Display slide numbers when 'slideNumber' is enabled */
.reveal .slide-number-pdf {
	display: block;
	position: absolute;
	font-size: 14px;
}

</script>


<script>
var style = document.createElement( 'style' );
style.type = 'text/css';
var style_script_id = window.location.search.match( /print-pdf/gi ) ? 'pdf-css' : 'paper-css';
var style_script = document.getElementById(style_script_id).text;
style.innerHTML = style_script;
document.getElementsByTagName('head')[0].appendChild(style);
</script>

    <script src="13-slides_files/header-attrs-2.4/header-attrs.js"></script>
    <link href="13-slides_files/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
    <link href="13-slides_files/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />
</head>
<body>
  <div class="reveal">
    <div class="slides">

<section>
    <h1 class="title">Transcription factor binding sites in Baker’s yeast</h1>
    <h2 class="author">M Hallett</h2>
    <h3 class="date">21/07/2020</h3>
</section>

<section id="where-we-are" class="slide level2">
<h2>Where we are</h2>
<figure class="image">
<p float="center">
<img src="images/explore_cycle.png" width="600"/>
</p>
<figcaption>
<small> From R for Datas Science, Chapter 6. </small>
</figcaption>
</figure>
</section>
<section id="models" class="slide level2">
<h2>Models</h2>
<ul>
<li><p>What is a model?</p></li>
<li><p>Why are they important?</p></li>
<li><p>How do we build a model?</p></li>
<li><p>How do we parameterize or train a model?</p></li>
<li><p>How do we evaluate its performance?</p></li>
<li><p>How do we validate a model?</p></li>
</ul>
</section>
<section id="models-1" class="slide level2">
<h2>Models</h2>
<ul>
<li><p>“All models are wrong but some are useful.” [Attributed to George Box]</p></li>
<li><p>“We all know that art is not truth. Art is a lie that makes us realize truth, at least the truth that is given us to understand. The artist must know the manner whereby to convince others of the truthfulness of his lies.” [Picasso, Pablo (1923), “Picasso speaks”]</p></li>
<li><p>“The models show hundreds of thousands of people are going to die and you know what I want to do? I want to come way under the model. The professionals did the models and I was never involved in a model. At least this kind of a model.” [An orange bouffon (Just before Nov. 3rd, 2020)]</p></li>
</ul>
</section>
<section id="transcription-factors-in-s.-cerevisiae" class="slide level2">
<h2>Transcription factors in S. cerevisiae</h2>
<ul>
<li><p>What are transcription factors?</p></li>
<li><p>How can we find out which proteins in S. cerevisiae are transription factors?</p></li>
<li><p>What are transcription factor binding sites (TFBSs)?</p></li>
<li><p>How can we locate them in S. cerevisiae?</p></li>
</ul>
</section>
<section id="what-is-a-transcription-factor" class="slide level2">
<h2>What is a transcription factor?</h2>
<ul>
<li><p>A transcription factor (TF) is a protein that regulates the rate of transcription (DNA <span class="math inline">\(\rightarrow\)</span> mRNA) by binding to specific regions of DNA.</p></li>
<li><p>TFs can work alone or in a coordinated fashion with other TFs, RNA molecules and proteins to regulate expression.</p></li>
<li><p>The TF binds to DNA through a DNA Binding Domain (DBD). The amino acid sequence of the DBD determines the specific nucleic acid sequence the TF binds.</p></li>
<li><p><a href="https://www.youtube.com/watch?v=MkUgkDLp2iE">TFs the Movie</a></p></li>
<li><p><a href="https://www.youtube.com/watch?v=Sd3imcsvXxI">Oxford lecture</a></p></li>
</ul>
</section>
<section id="transcription-factors-bind-dna" class="slide level2">
<h2>Transcription factors bind DNA</h2>
<figure class="image">
<p float="center">
<img src="images/tfs.jpg" width="700"/>
</p>
<figcaption>
<p><small> <a href="https://doi.org/10.1016/B978-0-12-386499-4.00005-7" class="uri">https://doi.org/10.1016/B978-0-12-386499-4.00005-7</a></p>
</small>
</figcaption>
</figure>
</section>
<section id="distinct-families-of-dbds-in-yeast" class="slide level2">
<h2>Distinct families of DBDs in Yeast</h2>
<ul>
<li>In yeast, (most) TFs bind DNA using degenerate versions of these five families of protein stuctures.</li>
</ul>
<figure class="image">
<p float="center">
<img src="images/dbds.png" width="400"/>
</p>
</figure>
</section>
<section id="transcription-factor-binding-sites" class="slide level2">
<h2>Transcription factor binding sites</h2>
<ul>
<li><p>The DBD binds DNA in a semi-specific manner. In other words, the location where the TF binds DNA via the DBD is characterized by a <em>degenerate motif</em>. There is no one single nucleic acid sequence necesssary and sufficieint for DNA binding. Rather there is a “family” of related nucleic acid sequences, a degenerate motif.</p></li>
<li><p>This is the binding motif for GCN4, a basic leucine ziper factor (bZIP) in yeast. You can see that some positions are highly determined (indices 7, 10, 12, 14) while others are variable (indices 7, 9, 11, 12, 15).</p></li>
</ul>
<figure class="image">
<p float="left">
<img src="images/count.png" width="400" height = "200"> <img src="images/bzip.png" width="400"/>
</p>
<figcaption>
<small> From <a href="http://jaspar.genereg.net/matrix/MA0303.1/">JASPAR</a>. </small>
</figcaption>
</figure>
</section>
<section id="transcription-factor-binding-prediction" class="slide level2">
<h2>Transcription factor binding prediction</h2>
<ul>
<li><p>Broadly speaking, there are two main questions.</p></li>
<li><p><strong>1. Find additional members of a known DNA binding motif.</strong></p></li>
<li><p>Here we typically know what the binding side for a TF looks like and we are asking to find other places in the genome of an organism(s) where this TF likely binds.</p></li>
<li><p>The motif is modelled, for example, using a position weight matrix (we’ll come back to this today) built from known examples of binding site (eg detemrined experimentally through SELEX experiments). Then this model is used to detect all occurences of the motfi in the genome.</p></li>
<li><p>Sometimes the search is performed not only with the target genome but several additional genomes spanning a considerable evolutionary distance. The principle underlying such <em>phylogenetic footprinting</em> approaches is that a nucleotide that is conserved across evolution is more likely to be function.</p></li>
<li><p>There are many software packages to do this including PhyloScan.</p></li>
<li><p>We will look at some of the fundamentals of this type of approach today.</p></li>
</ul>
</section>
<section id="transcription-factor-binding-prediction-1" class="slide level2">
<h2>Transcription factor binding prediction</h2>
<ul>
<li><p><strong>2. Discover a DNA binding motif for a specific TF.</strong></p></li>
<li><p>In this approach, we do not know the DNA binding motif for a TF and we want to learn it.</p></li>
<li><p>One way to do that would be to collect the promoters of a set of genes that we believe are all regulated by that TF. The idea is that the promoters should share a common motif that represents the location of binding for the TF.</p></li>
<li><p>The software package <a href="http://meme-suite.org/">MEME</a> is a classic example of this type of approach. It takes as input the DNA of a group of related sequences and identifies the sequence motifs that are most shared across this set of sequences. For example, you might give the promoters for a specif gene from several fungi. (MEME is really a suite of related appraoches that different in subtle but important ways.)</p></li>
<li><p><a href="http://homer.ucsd.edu/homer/introduction/basics.html">HOMER</a> is a conceptually-related but distinct approach. HOMER takes as input two sets of promoters. The first set might consist of promoters from genes that are highly expressed when the TF is activated compared to normal levels of the TF. The second set of promoters are from genes that are beleive not to be regulated by the TF. The intuition is that motifs that are enriched in the first but depleted in the second are more likely to be TF binding sites.</p></li>
</ul>
</section>
<section id="transcription-factors-in-s.-cerevisiae-1" class="slide level2">
<h2>Transcription factors in S. cerevisiae</h2>
<ul>
<li><p>How can we find out which proteins in S. cerevisiae are transription factors?</p></li>
<li><p><a href="http://geneontology.org/">Gene Ontology consortium</a></p></li>
<li><p><a href="https://www.yeastgenome.org/">Gene Ontology for S. cerevisiae (from SGD)</a></p></li>
<li><p><a href="http://amigo.geneontology.org/amigo/dd_browse">AmiGO</a>, a nice tool for visualy exploring GO</p></li>
</ul>
</section>
<section id="a-simple-model-for-tf-binding-sites" class="slide level2">
<h2>A simple model for TF binding sites</h2>
<p>(switch to notes)</p>
</section>
<section id="points-of-reflection" class="slide level2">
<h2>Points of reflection</h2>
<ul>
<li><p>Our model is very simple. Name some of aspects of DNA-protein interactions that are not captured by our model.</p></li>
<li><p>What other aspects of gene regulation are not well modelled by our approach?</p></li>
<li><p>How could we improve our mdoel to better reflect the bioogical reality of transcription factors, binding sites and gene regulation in general?</p></li>
<li><p>Pseudocounts are said to adjust our “prior belief” in certain events. Here what effect does adding the pseudocount to the position matrix have?</p></li>
</ul>
</section>
<section id="biol-480" class="slide level2">
<h2>BIOL 480</h2>
<p>© M Hallett, 2020 Concordia University</p>
<figure>
<img data-src="images/apple-touch-icon-114x114.png" alt="" /><figcaption>Computational Biology Concordia</figcaption>
</figure>
</section>
    </div>
  </div>

  <script src="13-slides_files/reveal.js-3.3.0.1/lib/js/head.min.js"></script>
  <script src="13-slides_files/reveal.js-3.3.0.1/js/reveal.js"></script>

  <script>

      // Full list of configuration options available at:
      // https://github.com/hakimel/reveal.js#configuration
      Reveal.initialize({
        // Display the page number of the current slide
        slideNumber: true,
        // Push each slide change to the browser history
        history: true,
        // Vertical centering of slides
        center: false,
        // Opens links in an iframe preview overlay
        previewLinks: true,
        // Transition style
        transition: 'convex', // none/fade/slide/convex/concave/zoom
        // Transition style for full page slide backgrounds
        backgroundTransition: 'default', // none/fade/slide/convex/concave/zoom

        menu: {
   
          numbers: true,
    
    
    
    
    
    
 
          custom: false,
          themes: false,
          transitions: false
        },



        chalkboard: {
          toggleNotesButton: true,
          theme: 'chalkboard',
        },

        keyboard: {
          67: function() { RevealChalkboard.toggleNotesCanvas() },    // toggle notes canvas when 'c' is pressed
          66: function() { RevealChalkboard.toggleChalkboard() }, // toggle chalkboard when 'b' is pressed
          46: function() { RevealChalkboard.clear() },    // clear chalkboard when 'DEL' is pressed
           8: function() { RevealChalkboard.reset() },    // reset chalkboard data on current slide when 'BACKSPACE' is pressed
          68: function() { RevealChalkboard.download() }, // downlad recorded chalkboard drawing when 'd' is pressed
        },

        // Optional reveal.js plugins
        dependencies: [
          { src: '13-slides_files/reveal.js-3.3.0.1/plugin/notes/notes.js', async: true },
          { src: '13-slides_files/reveal.js-3.3.0.1/plugin/search/search.js', async: true },
          { src: '13-slides_files/reveal.js-3.3.0.1/plugin/zoom-js/zoom.js', async: true },
          { src: '13-slides_files/reveal.js-3.3.0.1/plugin/chalkboard/chalkboard.js', async: true },
          { src: '13-slides_files/reveal.js-3.3.0.1/plugin/menu/menu.js', async: true },
        ]
      });
    </script>
  <!-- dynamically load mathjax for compatibility with self-contained -->
  <script>
    (function () {
      var script = document.createElement("script");
      script.type = "text/javascript";
      script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
      document.getElementsByTagName("head")[0].appendChild(script);
    })();
  </script>

<script>
  (function() {
    if (window.jQuery) {
      Reveal.addEventListener( 'slidechanged', function(event) {  
        window.jQuery(event.previousSlide).trigger('hidden');
        window.jQuery(event.currentSlide).trigger('shown');
      });
    }
  })();
</script>


  </body>
</html>
