<!DOCTYPE html>
<html lang="en-us"
      xmlns:og="http://ogp.me/ns#"
      xmlns:fb="https://www.facebook.com/2008/fbml">
<head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="author" content="Mike Hallett">
    <meta name="description" content="Explore and model biological data">

    <meta name="twitter:card" value="summary_large_image">
    <meta name="twitter:site" content="@hallettmichael">
    <meta name="twitter:title" content="Resources | BIOL 480/510/630: Bioinformatics">
    <meta name="twitter:creator" content="@hallettmichael">
    <meta name="twitter:image" content="/images/sys-small1.png">
    <meta name="twitter:description" content="Explore and model biological data" />

    <meta property="og:site_name" content="BIOL 480/510/630: Bioinformatics"/>
    <meta property="og:title" content="Resources | BIOL 480/510/630: Bioinformatics">
    <meta property="og:type" content="article" />
    <meta property="article:author" content="" />
    <meta property="og:url" content="/" />
    <meta property="og:image" content="/images/sys-small1.png">
    <meta property="og:description" content="Explore and model biological data" />

    <meta name="generator" content="Hugo 0.74.1" />    

    <title>Resources | BIOL 480/510/630: Bioinformatics</title>
    <link rel="canonical" href="/resources/">
    <link href="/theme/css/tufte.css" rel="stylesheet">
    <link href="/theme/css/latex.css" rel="stylesheet">
    <link href="/theme/css/custom.css" rel="stylesheet">

    <!--[if lt IE 9]>
    <script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
    <script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
    <![endif]-->

    <style type="text/css">
        .wrap a, .wrap a:visited {
            color: #1663BA;
        }

        .wrap a:hover {
            color: #398AE6;
        }

        .wrap p > a {
            color: #1663BA;
        }

        article .sidenote-number:after, article .sidenote:before {
            color: #1663BA;
        }

        article .sidenote:before {
            color: #1663BA;
        }

        @media (max-width: 760px) {
            label.margin-toggle:not(.sidenote-number) {
                color: #1663BA;
            }
        }

        .contrast a, .contrast a:visited {
            color: #1663BA; 
        }

        header > nav a {
            color: #1663BA;
        }

        .credits a {
            color: #1663BA;
        }

        .wrap .tag {
            background: #1663BA;
        }
    </style>

    <link rel="shortcut icon" href="/favicon.ico" />
    <link rel="apple-touch-icon-precomposed" sizes="57x57" href="/images/apple-touch-icon-57x57.png" />
    <link rel="apple-touch-icon-precomposed" sizes="114x114" href="/images/apple-touch-icon-114x114.png" />
    <link rel="apple-touch-icon-precomposed" sizes="72x72" href="/images/apple-touch-icon-72x72.png" />
    <link rel="apple-touch-icon-precomposed" sizes="144x144" href="/images/apple-touch-icon-144x144.png" />
    <link rel="apple-touch-icon-precomposed" sizes="120x120" href="/images/apple-touch-icon-120x120.png" />
    <link rel="apple-touch-icon-precomposed" sizes="152x152" href="/images/apple-touch-icon-152x152.png" />
    <link rel="icon" type="image/png" href="/images/favicon-32x32.png" sizes="32x32" />
    <link rel="icon" type="image/png" href="/images/favicon-16x16.png" sizes="16x16" />
    <meta name="application-name" content="BIOL 480/510/630: Bioinformatics"/>
    <meta name="msapplication-TileColor" content="#FFFFFF" /> 
    <meta name="msapplication-TileImage" content="/images/mstile-144x144.png" />

    
    <script src="//use.typekit.net/vpe6zfl.js"></script>
    <script>try{Typekit.load({ async: true });}catch(e){}</script>

    <link href="//maxcdn.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css" rel="stylesheet">


 

</head>

<body>
<div class="wrap">
    <header>
        <nav class="group logo-wrap">
            <a href="/" class="logo"><img src="/images/sys-small1.png" alt="Data Science in Biology"></a>
            <h1 class="websitetitle"><a href="/">BIOL 480/510/630: Bioinformatics</a></h1>
        </nav>
        <nav class="bottom-group">
            
            <a href="/overview/">Overview</a>
            
            <a href="/syllabus/">Syllabus</a>
            
            <a href="/resources/">Resources</a>
            
            <a href="/assignments/">Assignments</a>
            
            <a href="/projects/">Projects</a>
            
        </nav>
    </header>
 

    <article class="group">
    <h1>Resources</h1>

    
<!-- BLOGDOWN-HEAD -->
<script src="/rmarkdown-libs/header-attrs/header-attrs.js"></script>
 
<!-- /BLOGDOWN-HEAD -->

<h2>Contents</h2>
<div id="TOC">
<ul>
<li><a href="#information-technologies-for-life-science-labs">Information technologies for life science labs</a></li>
<li><a href="#modern-lab-notebooks">Modern lab notebooks</a></li>
<li><a href="#time-management">Time management</a></li>
<li><a href="#resource-management">Resource management</a></li>
<li><a href="#community-support">Community Support</a></li>
<li><a href="#scientific-literature">Scientific literature</a></li>
<li><a href="#bioinformatic-resources">Bioinformatic resources</a></li>
<li><a href="#r-rstudio-and-tidyverse-resources">R, RStudio and tidyverse resources</a></li>
<li><a href="#how-to-select-the-appropriate-chart-type">How to select the appropriate chart type</a></li>
<li><a href="#colors-and-fonts">Colors and fonts</a></li>
<li><a href="#markdown">Markdown</a></li>
<li><a href="#citations-and-bibliography">Citations and bibliography</a></li>
</ul>
</div>

<h3 id="information-technologies-for-life-science-labs">Information technologies for life science labs</h3>
<ul>
<li><p><a href="https://drive.google.com"><i class="fa fa-google"></i> Google Drive</a> An excellent and convenient collection of tools (documents, images, spreadsheets, presentations) with lots of storage space. This is cloud-based so teams can work on documents simultaneously.</p></li>
<li><p><a href="https://www.zotero.org/"><i class="fas fa-book"></i> Zotero</a> It is really important to develop your library of bibliographic references, but it can certainly be challenging to keep all the references organized. Also, it is time consming to manually enter all the details of reference. There are many good tools out there; we tend to use Zotero as it is well integrated with Google Docs (for text files).</p></li>
<li><p><a href="https://colab.research.google.com"><i class="fa fa-google"></i> Google Colab</a> A promising cloud-based approach for a team to program together. It doesn’t currently have great support for <span class="math inline">\({\tt R}\)</span> but perhaps in the near future(?). It is a good option for <span class="math inline">\({\tt Python}\)</span> currently.</p></li>
<li><p><a href="https://cloud.google.com/"><i class="fa fa-google"></i> Google Cloud</a> Whereas Google Drive is for storing files and developing documents, Google Cloud provides as much CPU, GPU, and TPU power as you want or can afford. There are other options including <span class="math inline">\({\tt AWS}\)</span> from Amazon and <span class="math inline">\({\tt Azure}\)</span> from Microsoft, amongst others.</p></li>
</ul>
<p>Note that RStudio Cloud is a kind of restricted combination of Google Drive, Colab and Cloud. It suffices for the teething stage while learning bioinformatics and biodata science.</p>
<ul>
<li><p><a href="https://slack.com/"><i class="fa fa-slack"></i> Slack</a> We will use this non-stop in the course. There are alternatives including open source efforts but for various reasons we choose <span class="math inline">\({\tt Slack}\)</span> for all of our laboratory-related work. Any of these tools are far far superior than email (yikes).</p></li>
<li><p><a href="https://git-scm.com/"><i class="fa fa-git"></i> Git</a> The course will touch on GIT, a very important piece of software that provides version control. GIT tracks all the files in a project as they are modified by different members of a team. It allows you to see the entire history of all modifications to any file and to “roll back” the files if necessary. Once your project is stored in GIT, it’s basically impossible to lose it. Our entire course is maintained under GIT and you have full access to it through GitHub (see next entry).</p></li>
<li><p><a href="https://github.com/"><i class="fa fa-github"></i> GitHub</a> GitHub is a cloud resource providing facilities to house and share code persistently and consistently. It is tightly assocaited with GIT. There are many alternatives. We use both GitHub and <a href="bitbucket.org"><i class="fa fa-bitbucket"></i>BitBucket</a>.</p></li>
</ul>
<h3 id="modern-lab-notebooks">Modern lab notebooks</h3>
<ul>
<li><p>These are interesting short articles on different approaches towards keeping lab notebooks, and the pros and cons of electronic laboratory notebooks (ELNs) versus traditional paper notebooks. <a href="https://www.sciencemag.org/careers/2019/09/how-keep-lab-notebook">E Pain (2019) Science</a> and <a href="https://www.nature.com/articles/d41586-018-05895-3">R Kwok (2018) Nature</a>.</p></li>
<li><p><a href="https://www.labtwin.com/"><i class="fa fa-wrench"></i> LabTwin</a> We recently installed LabTwin in our wet labs. LabTwin can be activated by voice via Siri, Alexa etc “hands-free” in the lab. It uses a robust artificial intelligence-supported interface to help you record your notes, display protocols, coordinate experiments and provide information “on the fly”.</p></li>
<li><p>I personally use a tool called <a href="https://www.goodnotes.com"><i class="fa fa-goodnotes"></i> Goodnotes</a>. Others in my lab use <a href="https://www.evernote.com"><i class="fa fa-evernote"></i> Evernote</a>. Although I haven’t used it, I hear very good things about
<a href="https://notion.so/"><i class="fa fa-wrench"></i> Notion</a>. This is software that really integrates many tools into one coherent framework. If everyone in a group is using the software, it is very easy to share notes, images, calendars, and other common information. As it is cloud based, the information is available anywhere to all members of the group.</p></li>
</ul>
<h3 id="time-management">Time management</h3>
<p>There are many software packages for designing, executing and managing large projects that involve many people. Most of these packages are built for companies. However several are particularly useful for modern life science labs.</p>
<ul>
<li><p><a href="https://trello.com/"><i class="fa fa-wrench"></i> Trello</a> This is a project planning tool that is quite handy. It is not life science specific and could be used for almost any time of proect or event that is complicated to plan. Some members of my lab use it.</p></li>
<li><p><a href="https://labscrum.org/"><i class="fa fa-wrench"></i> Labscrum</a> This is really more of a technique than softare per se for conducting academic scientific research. It is a way of organizing a group, and provides a method to assist in tracking progress of projects and researchers. Our group does not use this, but it looks interesting.</p></li>
<li><p><a href="https://toggl.com/"><i class="fa fa-power-off"></i> Toggl</a> This is an app that allows you to manage your time. Basically, you can keep track of how long you spend on different tasks during the day. Many private consultants use this type of tool to record how long they spend on a client’s project for billing purposes. It is also useful if you have trouble managing your study and research time.</p></li>
</ul>
<h3 id="resource-management">Resource management</h3>
<p>Especially with wet labs, a consistent effort is needed to order, track and invoice reagents, kits, and other expendables. Moreover, there is a need to precisely and accurately handle samples. For instance, as we process samples donated by cancer patients, we have a moral obligation to ensure that the material is not lost or mislabeled. Our lab certainly finds it challenging to maintain a “chain of custody”. We have experience with several software systems.</p>
<ul>
<li><p><a href="https://www.quartzy.com/"><i class="fa fa-wrench"></i> Quartzy</a> and <a href="https://www.benchling.com/"><i class="fa fa-benchling"></i> Benchling</a> are both nice software systems, each with their pros and cons.</p></li>
<li><p><a href="https://snipeitapp.com/"><i class="fa fa-qrcode"></i> Snipe-It</a> We recently installed this system as a way to generate QR- (or classic bar-) codes. Basically, you can print out QR code stickers and associate each QR code with one item (reagent, buffer, sample, computer, eraser) in your lab. You need only a cell phone with a camera. You can purchase cheap special sticker paper that survives autoclaves and -80 freezers.</p></li>
</ul>
<h3 id="community-support">Community Support</h3>
<p>Finding on-line communities to answer questions as they arise is super imporant.</p>
<ul>
<li><p><a href="https://stackoverflow.com/"><i class="fa fa-stack-overflow"></i> Stackoverflow</a> This is an amazingly useful website that you will discover many times when asking questions especially with respect to computers and IT issues. It is community driven, meaning that people voluntarily answer each other’s questions and provide advice from the very mundane to the most technical challenges.</p></li>
<li><p><a href="https://www.biostars.org/"><i class="fa fa-info-circle"></i> Biostars</a> If your question is bioinformatic in nature (eg what software to use? how does a particular piece of software work? how do I design my experiment optimally? etc.), the Biostars website and community is excellent.</p></li>
<li><p><a href="https://community.rstudio.com/"><i class="fa fa-rebel"></i> RStudio Community</a> If your questions are about operating RStudio or its products (incl. Shiny, blogdown) specific. Not generally the place for assistance with coding in <span class="math inline">\({\tt R}\)</span>.</p></li>
</ul>
<h3 id="scientific-literature">Scientific literature</h3>
<p>It is difficult to keep up to date with new publications in your area(s) of interest. In addition to standard Google searches, I use some of the following strategy each day.</p>
<ul>
<li><p><a href="https://pubmed.ncbi.nlm.nih.gov/"><i class="fa fa-search"></i> NCBI’s PubMed</a> Provides access to almost all scientific publications.<span><label for="sn-0" class="margin-toggle sidenote-number"></label><input type="checkbox" id="sn-0" class="margin-toggle"/><span class="sidenote">Note that Concordia has subscriptions to many (non-open-source) journals. You need to connect to the Concordia VPN (<a href="https://www.concordia.ca/content/dam/concordia/docs/IITS/VPN/VPN_AnyConnect_Windows_v2.pdf">instructions here</a>).<br />
<br />
</span></span> The NCBI has a tool where you can receive daily updates (by email) of all abstracts containing specific keywords.<span><label for="sn-1" class="margin-toggle sidenote-number"></label><input type="checkbox" id="sn-1" class="margin-toggle"/><span class="sidenote">Each morning I troll through these emails for relevant papers.<br />
<br />
</span></span></p></li>
<li><p><a href="https://www.biorxiv.org/">bioRxiv preprints</a> It is becoming increasingly common to submit manuscripts before publication to the preprint servers such as <span class="math inline">\({\tt bioRxiv}\)</span>. In fact, my group finds that there is quite often a year’s delay between when a preliminary (unrefereed) version of a manuscript appears as a preprint and when it appears in a journal in its final version.<span><label for="sn-2" class="margin-toggle sidenote-number"></label><input type="checkbox" id="sn-2" class="margin-toggle"/><span class="sidenote">A problem with preprint servers is that it is like drinking from a fire hose. It’s hard to zero in on the papers you want to see.<br />
<br />
</span></span></p></li>
<li><p><a href="https://twitter.com/hallettmichael"><i class="fa fa-twitter"></i> Twitter specifically #academictwitter</a> In addition to a platform that allows questionable right wing politicians to tweet questionable policy decisions, Twitter cannot be ignored as a valuable tool for following your scientific community.<span><label for="sn-3" class="margin-toggle sidenote-number"></label><input type="checkbox" id="sn-3" class="margin-toggle"/><span class="sidenote">I spend a considerable amount of micro-breaks following links to tweets of interesting look papers. Also, I find that Twitter allows you to expand out a bit and see groups/papers you might otherwise not run into.<br />
<br />
</span></span> I highly recommend that each student open a Twitter account for their “professional” purposes and follow members of your community. This includes journals and preprint servers who regularly tweet new papers.<span><label for="sn-4" class="margin-toggle sidenote-number"></label><input type="checkbox" id="sn-4" class="margin-toggle"/><span class="sidenote">I often rely on journals to tweet new papers rather than going to the journal’s webpage or having them send me an email (yikes).<br />
<br />
</span></span></p></li>
<li><p><a href="https://scholar.google.com/"><i class="fa fa-google"></i> Google Scholar</a> Not surprisingly, this is a very powerful search engine for academic groups and publications. I recommend every student to establish a profile at Google Scholar.</p></li>
<li><p>Otherwise, I greatly rely on my group to post interesting articles to our <span class="math inline">\({\tt Slack}\)</span> workspace and to give mini-presentations of potentially interesting papers in our lab meetings.</p></li>
</ul>
<h3 id="bioinformatic-resources">Bioinformatic resources</h3>
<ul>
<li><p><a href="https://www.ncbi.nlm.nih.gov/"><i class="fa fa-database"></i> National Center for Biotechnology Information (NCBI)</a> A huge resource from the USA that houses a vast spectrum of biological data and datasets, and bioinformatic tools.</p></li>
<li><p><a href="https://www.embl.org/"><i class="fa fa-database"></i> European Molecular Biology Labs (EMBL)</a> and <a href="https://www.ebi.ac.uk/">EMBL-European Bioinformatics Institute (EBI)</a> are the European analogs to the NCBI with similar mandates and scope.</p></li>
<li><p><a href="https://jgi.doe.gov/"><i class="fa fa-database"></i> Joint Genomes Institute (JGI)</a>
This is US Department of Energy, Lawrence Berkeley lab intiative that has really excelled recently in many different domains. My lab uses tools such as <a href="https://jgi.doe.gov/data-and-tools/mycocosm/">MycoCosm</a> often for fungal (microbiome) studies.</p></li>
</ul>
<p>There are many other databases, often with very specific mandates. For example,
many organisms have their own database <a href="https://www.yeastgenome.org/">Saccharomyces cerevisiae Genome DataBase (SGD)</a> and many human diseases have specialized resources. For example, these are two important cancer resources:</p>
<ul>
<li><p><a href="https://www.cancer.gov/about-nci/organization/ccg/research/structural-genomics/tcga"><i class="fa fa-database"></i> The Cancer Genome Atlast (TCGA) program</a>. TCGA, which is USA based, contains high-throughput profiles for several large patient cohorts across different cancer types. The availability of multi-modal high-throughput profiles conducting in a uniform manner across these cohorts opens up many analysis opportunities.</p></li>
<li><p><a href="https://dcc.icgc.org/"><i class="fa fa-database"></i> International Cancer Genome Consortium (ICGC)</a> is similar to TCGA but an international consortium across more types of cancers.</p></li>
</ul>
<h3 id="r-rstudio-and-tidyverse-resources">R, RStudio and tidyverse resources</h3>
<p>There is now likely an uncountably infinite number<span><label for="sn-5" class="margin-toggle sidenote-number"></label><input type="checkbox" id="sn-5" class="margin-toggle"/><span class="sidenote">well perhaps not <strong>that</strong> many but …<br />
<br />
</span></span> of resources for learning R, RStudio and assocaiated packages and extensions.</p>
<ul>
<li><p><a href="https://www.r-project.org/">The R Language</a> This is the main website for the core R language. There is a tonne of useful information including manuals, books and other technical information. You can download (for free) R from this site (if you would like to set up R independently of RStudio Cloud).</p></li>
<li><p><a href="https://cran.r-project.org/doc/manuals/r-release/R-intro.html">The R Manual</a> If you would like to learn the basics of R in a more traditional approach (compared to the data science focus used in this course), this manual is a good place to start.</p></li>
<li><p><a href="https://www.rstudio.com/resources/cheatsheets/">R and RStudio cheat sheets</a>: A large collection of simple cheat sheets for RStudio, <code>ggplot2</code>, and other R-related things. RStudio also has an excellent series of <a href="https://rstudio.com/resources/webinars/">webinars</a> covering almost all aspects of R and RStudio. I highly recommend these easy to follow videos.</p></li>
<li><p><a href="http://www.bioconductor.org/">Bioconductor</a> is a collection of packages and tools for the biosciences. It is an open source effort. Although not strictly limited to R, Bioconductor is certainly well integrated with R.</p></li>
<li><p><a href="https://www.datacamp.com/">DataCamp</a> has excellent R tutorials where lectures are intermixed with hands-on programming exercises similar to the <span class="math inline">\({\tt LearnR}\)</span> modules you will experience in this course. You might consider trying the <span class="math inline">\({\tt tidyverse}\)</span> tutorial that is very high quality in my opinion. There are many competitors to DataCamp, some of which are free. DataCamp allows for limited free access.</p></li>
<li><p><a href="http://stat545.com/">Stat 545</a>: <a href="https://twitter.com/JennyBryan">Dr. Jenny Bryan</a> at the University of British Columbia has an entire introductory course in R, visualization, and data analysis online.</p></li>
</ul>
<h3 id="how-to-select-the-appropriate-chart-type">How to select the appropriate chart type</h3>
<p><span><label for="sn-6" class="margin-toggle">&#8853;</label><input type="checkbox" id="sn-6" class="margin-toggle"/><span class="marginnote"> <a href="http://www.datavizcatalogue.com/"><img src="/images/reference/dataviz-catalogue.png" alt="The Data Visualisation Catalogue" /></a> <em>The Data Visualisation Catalogue</em><br />
<br />
</span></span> I find the collections below very useful: when I want to make a plot, I typically search through hundreds of images to find the most suitable/exciting.<span><label for="sn-7" class="margin-toggle sidenote-number"></label><input type="checkbox" id="sn-7" class="margin-toggle"/><span class="sidenote">Thank you to <a href="https://www.andrewheiss.com">A Heiss</a> for compiling this selection.<br />
<br />
</span></span> Here are some of the best:</p>
<ul>
<li><a href="http://www.datavizcatalogue.com/">The Data Visualisation Catalogue</a>: Descriptions, explanations, examples, and tools for creating 60 different types of visualizations.</li>
<li><a href="http://datavizproject.com/">The Data Viz Project</a>: Descriptions and examples for 150 different types of visualizations. Also allows you to search by data shape and chart function (comparison, correlation, distribution, geographical, part to whole, trend over time, etc.).</li>
<li><a href="http://chartmaker.visualisingdata.com/">The Chartmaker Directory</a>: xamples of how to create 51 different types of visualizations in 31 different software packages, including Excel, Tableau, and R.</li>
<li><a href="https://www.r-graph-gallery.com/">R Graph Catalog</a>: R code for 124 ggplot graphs.</li>
<li><a href="http://annkemery.com/essentials/">Emery’s Essentials</a>: Descriptions and examples of 26 different chart types.</li>
</ul>
<h3 id="colors-and-fonts">Colors and fonts</h3>
<ul>
<li><a href="http://colorbrewer2.org/">ColorBrewer</a>: Sequential, diverging, and qualitative color palettes that take accessibility into account.</li>
<li><a href="https://fonts.google.com/">Google Fonts</a>: Huge collection of free, well-made fonts.</li>
<li><a href="https://www.reliablepsd.com/ultimate-google-font-pairings/">The Ultimate Collection of Google Font Pairings</a>: A list of great, well-designed font pairings from all those fonts hosted by Google (for when you’re looking for good contrasting or complementary fonts).</li>
</ul>
<h3 id="markdown">Markdown</h3>
<p>The slides and website for this course were developed in Markdown, specifically a dialect deigned for R (<span class="math inline">\({\tt RMarkdown}\)</span>). Fluency in Markdown is a useful skill to have as more and more tools and information migrates to the cloud.</p>
<ul>
<li><span><label for="sn-8" class="margin-toggle">&#8853;</label><input type="checkbox" id="sn-8" class="margin-toggle"/><span class="marginnote"> <img src="/images/reference/markdown.png" alt="Markdown logo" /><br />
<br />
</span></span> <a href="http://plain-text.co/">The Plain Person’s Guide to Plain Text Social Science</a>: A comprehensive explanation and tutorial about why you should write data-based reports in Markdown.</li>
<li><a href="https://www.markdowntutorial.com/">Markdown tutorial</a>: An interactive tutorial to practice using Markdown.</li>
<li><a href="http://packetlife.net/media/library/16/Markdown.pdf">Markdown cheatsheet</a>: Useful one-page reminder of Markdown syntax.</li>
</ul>
<h3 id="citations-and-bibliography">Citations and bibliography</h3>
<p><span><label for="sn-9" class="margin-toggle">&#8853;</label><input type="checkbox" id="sn-9" class="margin-toggle"/><span class="marginnote"> You can open the file in <a href="http://bibdesk.sourceforge.net/">BibDesk on macOS</a>, <a href="http://www.jabref.org/">JabRef on Windows</a>, or <a href="https://www.zotero.org/">Zotero</a> or <a href="https://www.mendeley.com/">Mendeley</a> online. I personnally use Zotero with Google Docs.<br />
<br />
</span></span> You can <a href="/bib/references.bib">download a BibTeX file</a> of all the non-web-based readings in the course.</p>


    </article>

    <footer>
        <hr class="slender">
        <div class="course-info">
        <span class="course-name">
            BIOL 480/510/630: Bioinformatics (Fall 2020)
        </span><br>
        <span>
            <em><a href="https://www.concordia.ca">Concordia University</a></em> | 
            <em><a href="https://www.concordia.ca/artsci/biology.html">Department of Biology | CSFG | CASB</a></em>
        </span>
        </div>

        <ul class="footer-links">
            <li>
            <a href="https://mikehallett.science"><i class="fa fa-user" aria-hidden="true"></i> Mike Hallett</a>
            </li>

            <li>
            <a href="mailto:hallett.mike.t@gmail.com"><i class="fa fa-envelope" aria-hidden="true"></i> gmail</a>
            </li>

            <li>
            <i class="fa fa-calendar" aria-hidden="true"></i> TuTh
            </li>

            <li>
            <i class="fa fa-clock-o" aria-hidden="true"></i> 13:15-14:30
            </li>

            <li>
                <a href=""><i class="fa fa-university" aria-hidden="true"></i> Zoom</a>
            </li>
            

            <li>
            <a href="https://twitter.com/hallettmichael"><i class="fa fa-twitter" aria-hidden="true"></i></a> 
            </li>

            <li>
            <a href="https://github.com/https://github.com/hallettmiket/hallettmiket.github.io"><i class="fa fa-github" aria-hidden="true"></i></a>
            </li>
            <span class="fake-li"><i class="fa fa-slack" aria-hidden="true"></i> <a href="https://biol480-18-concordia.slack.com/">Slack</a></span><br>

            
        </ul>

        <div class="credits">
        <span>All content licensed under a <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/">Creative Commons</a>.
        Content © 2020 <a href="https://mikehallett.science">mike hallett</a>
        </span><br><br>
<br><br>
        <span>We thank <a href="https://www.andrewheiss.com/">A Heiss</a> for the modified Tufte scheme.</span><br><br>

        <span>This site modified the <a href="https://github.com/andrewheiss/ath-tufte-hugo">ath-tufte-hugo theme</a> in <a href="https://bookdown.org/yihui/blogdown/">blogdown</a> and <a href="https://gohugo.io/">hugo</a>.</span>
        </div>
    </footer>

    
    <script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      tex2jax: {
        skipTags: ['script', 'noscript', 'style', 'textarea', 'pre']
      }
    });
    </script>
    <script async type="text/javascript"
      src="https://cdn.bootcss.com/mathjax/2.7.0/MathJax.js?config=TeX-MML-AM_CHTML">
    </script>


</div>
</body>
</html>

